src/components/form-elements/Select.jsx

```diff
@@ -115,11 +142,16 @@ const Select = (props: SelectPropsType) => {
     },
     className
   );
-  const optionsElements = options.map(({value, text}) => (
-    <option key={value} value={value}>
-      {text}
-    </option>
-  ));
+
+  const optionsElements =
+    (options.length && options.map(getOptionElement)) ||
+    (groupedOptions.length &&
+      groupedOptions.map(({options, label}) => (
+        <optgroup key={label} label={label}>
+          {options.map(getOptionElement)}
+        </optgroup>
+      ))) ||
+    [];
```

Comment:
Wdyt about splitting this into consts to increase readability? 

```suggestion
  const flatOptionsElements = (options.length && options.map(getOptionElement));
  const groupedOptionsElements = (groupedOptions.length &&
      groupedOptions.map(({options, label}) => (
        <optgroup key={label} label={label}>
          {options.map(getOptionElement)}
        </optgroup>);
        
  const optionsElements = flatOptionsElements || groupedOptionsElements || [];
```

Reply:
This was my first solution. I did not find it more readable at that moment, it just enforces to create 1 extra unnecessary variable. But up to you, I do not mind changing it back.